tests/plugins/test_generate_maven_metadata.py

```diff
@@ -0,0 +1,259 @@
+DEFAULT_ARCHIVES = [
+    ARCHIVE_JAXB_SUN_POM,
+    ARCHIVE_JAXB_SUN_JAR,
+    ARCHIVE_JAXB_JAVADOC_SUN_JAR,
+    ARCHIVE_JAXB_GLASSFISH_POM,
+    ARCHIVE_JAXB_GLASSFISH_JAR,
+    ARCHIVE_JAXB_JAVADOC_GLASSFIX_JAR,
+    ARCHIVE_SHA1,
+    ARCHIVE_SHA256,
+]
+
+
+class MockSource(object):
+    def __init__(self, tmpdir):
+        tmpdir = str(tmpdir)
+        self.dockerfile_path = os.path.join(tmpdir, 'Dockerfile')
+        self.path = tmpdir
+
+    def get_build_file_path(self):
+        return self.dockerfile_path, self.path
+
+
+class MockedPathInfo(object):
+    def __init__(self, topdir=None):
+        self.topdir = topdir
+
+    def mavenbuild(self, build):
+        return '{topdir}/packages/{name}/{version}/{release}/maven'.format(topdir=self.topdir,
+                                                                           **build)
+
+    def mavenfile(self, maveninfo):
+        return '{group_id}/{artifact_id}/{version}/{filename}'.format(**maveninfo)
+
+
+def mock_koji_session(koji_proxyuser=None, koji_ssl_certs_dir=None,
+                      koji_krb_principal=None, koji_krb_keytab=None,
+                      build_info=None, archives=None):
+    if not build_info:
+        build_info = DEFAULT_KOJI_BUILD
+    if not archives:
+        archives = DEFAULT_ARCHIVES
+
+    flexmock(koji, PathInfo=MockedPathInfo)
+
+    session = flexmock()
+
+    (flexmock(koji)
+     .should_receive('ClientSession')
+     .once()
+     .and_return(session))
+
+    def mock_get_build(nvr):
+        if nvr == DEFAULT_KOJI_BUILD['nvr']:
+            return DEFAULT_KOJI_BUILD
+        else:
+            return None
+
+    (session
+     .should_receive('getBuild')
+     .replace_with(mock_get_build))
+
+    (session
+     .should_receive('listArchives')
+     .and_return(archives))
+
+    (session
+     .should_receive('krb_login')
+     .and_return(True))
+    return session
+
+
+def mock_fetch_artifacts_by_nvr(tmpdir, contents=None):
+    if contents is None:
+        contents = dedent("""\
+            - nvr: com.sun.xml.bind.mvn-jaxb-parent-2.2.11.4-1
+            """)
+
+    with open(os.path.join(tmpdir, REPO_FETCH_ARTIFACTS_KOJI), 'w') as f:
+        f.write(contents)
+        f.flush()
+
+
+def mock_nvr_downloads(build_info=None, archives=None, overrides=None):
+    if not build_info:
+        build_info = DEFAULT_KOJI_BUILD
+    if not archives:
+        archives = DEFAULT_ARCHIVES
+    if not overrides:
+        overrides = {}
+
+    pi = koji.PathInfo(topdir=KOJI_ROOT)
+
+    for archive in archives:
+        url = pi.mavenbuild(build_info) + '/' + pi.mavenfile(archive)
+        # Use any overrides for this archive ID
+        archive_overrides = overrides.get(archive['id'], {})
+        status = archive_overrides.get('status', 200)
+        body = archive_overrides.get('body', archive['filename'] + archive['group_id'])
+        responses.add(responses.GET, url, body=body, status=status)
+
+
+@responses.activate  # noqa
```

Comment:
Is `# noqa` necessary? `responses` is used through the codebase, `# noqa` is not used for this decorator at other place.

Reply:
Removed it. Not sure what it does. I thought it was some sort of note to  QA.